File: server/src/Models/DMToolkitModels/Definitions/LineageDefinition.cs
namespace DMToolkit.Models.Definitions;

public class LineageDefinition : IDefinition
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public ICollection<FeatDefinition> FeatDefinitions { get; set; } = new List<FeatDefinition>();
    public string SpeciesDefinitionId { get; set; } = string.Empty;
    public SpeciesDefinition? SpeciesDefinition { get; set; } = null;
}

File: web-client/src/lib/types/dm-tool-types/definitions/lineageDefinition.ts
import type { FeatDefinition } from "./featDefinition";

export interface LineageDefinition {
    id: string;
    name: string;
    description: string;
    featDefinitions: FeatDefinition[];
}

export interface LineageDefinitionDto {
    id: string;
    name: string;
    description: string;
    featDefinitionIds: string[];
}